fix for upstream for building with newer gcc versions

r339 | zhanyong.wan | 2009-11-12 21:54:23 -0500 (Thu, 12 Nov 2009) | 2 lines

...; makes gtest-param-util-generated.h conform to the C++ standard (by Zhanyong Wan).

Index: include/gtest/internal/gtest-param-util-generated.h
===================================================================
--- include/gtest/internal/gtest-param-util-generated.h	(revision 338)
+++ include/gtest/internal/gtest-param-util-generated.h	(revision 339)
@@ -53,6 +53,21 @@
 #if GTEST_HAS_PARAM_TEST
 
 namespace testing {
+
+// Forward declarations of ValuesIn(), which is implemented in
+// include/gtest/gtest-param-test.h.
+template <typename ForwardIterator>
+internal::ParamGenerator<
+    typename ::std::iterator_traits<ForwardIterator>::value_type> ValuesIn(
+        ForwardIterator begin, ForwardIterator end);
+
+template <typename T, size_t N>
+internal::ParamGenerator<T> ValuesIn(const T (&array)[N]);
+
+template <class Container>
+internal::ParamGenerator<typename Container::value_type> ValuesIn(
+    const Container& container);
+
 namespace internal {
 
 // Used in the Values() function to provide polymorphic capabilities.
